server/api.js

```diff
@@ -50,19 +50,26 @@ const botUser = new Bot({
   spotifyApi: spotifyApi
 });
 
-let users = [
-  {
-    user: botUser.toJSON()
-  }
-];
+let users = [botUser.toJSON()];
 
 let globalSocket = null;
 
+let globalIo = null;
+
 const queueManager = new QueueManager({
   onPlay: () => {
     const { track, user } = queueManager.getPlayingContext();
-    globalSocket && globalSocket.emit('play track', track, user);
-    globalSocket && globalSocket.broadcast.emit('play track', track, user);
+    // if one user logs in on multiple tabs, just send 'play track' message to one tab,
+    // but need to send 'update now playing' to other tabs
+    users.forEach(u => {
+      u.socketIdArray.forEach((socketId, index) => {
+        if (index === 0) {
```

Comment:
The user playing the track doesn't necessarily need to be in the first position of the array, right?

Reply:
I don't quite understand, what is the user playing the track? Does it mean the user who added the song?

The `socketIdArray` of the `user` is the user's all related sockets. I just need to pick one (it's easy to pick the first one) to send `play track` message, and send `update now playing` to other sockets. Am I wrong?

Reply:
Any update, maybe you're busy.

Reply:
My bad, sorry! I don't know why I thought that the index was the index for the user, but it's that index of the socket